extensions/blocks/membership-button/view.js

```diff
@@ -0,0 +1,57 @@
+/* global tb_show */
+
+/**
+ * Internal dependencies
+ */
+import './view.scss';
+const name = 'membership-button';
+const blockClassName = 'wp-block-jetpack-' + name;
+
+function activateSubscription( block, blogId, planId, poweredText ) {
+	block.addEventListener( 'click', () => {
+		tb_show(
+			null,
+			'https://subscribe.wordpress.com/memberships/?blog=' +
+				blogId +
+				'&plan=' +
+				planId +
+				'TB_iframe=true&height=600&width=400',
+			null
+		);
+		const tbWindow = document.querySelector( '#TB_window' );
+		tbWindow.classList.add( 'jetpack-memberships-modal' );
+		const footer = document.createElement( 'DIV' );
+		footer.classList.add( 'TB_footer' );
+		footer.innerHTML = poweredText;
+		tbWindow.appendChild( footer );
+	} );
+}
+
+const initializeMembershipButtonBlocks = () => {
+	const mailchimpBlocks = Array.from( document.querySelectorAll( '.' + blockClassName ) );
```

Comment:
Make sure to add `wp-polyfill` as a dependency because of `Array.from` which doesn't work in IE11.

Reply:
I'm going to change it to `Array.prototype.slice.call` (https://davidwalsh.name/nodelist-array) since that sounds better than introducing another dependency :)

Reply:
Sure, either way works for now. We (Gutenpack) will soonish enfoce `wp-polyfill` dependency on all blocks in Jetpack just because currently this is too human-error prone since the build process doesn't polyfill like developers might expect.